src/code/z_demo.c

```diff
@@ -391,176 +448,189 @@ void func_80064824(PlayState* play, CutsceneContext* csCtx, CsCmdBase* cmd) {
                 }
             }
             break;
-        case 28:
-            play->unk_11DE9 = true;
+
+        case CS_MISC_HALT_ALL_ACTORS:
+            play->haltAllActors = true;
             break;
-        case 29:
-            play->unk_11DE9 = false;
+
+        case CS_MISC_RESUME_ALL_ACTORS:
+            play->haltAllActors = false;
             break;
-        case 30:
-            Flags_SetEnv(play, 3);
+
+        case CS_MISC_SET_FLAG_3:
+            CutsceneFlags_Set(play, 3);
             break;
-        case 31:
-            Flags_SetEnv(play, 4);
+
+        case CS_MISC_SET_FLAG_4:
+            CutsceneFlags_Set(play, 4);
             break;
-        case 32:
-            if (sp3F != 0) {
+
+        case CS_MISC_SANDSTORM:
+            if (isFirstFrame) {
                 play->envCtx.sandstormState = SANDSTORM_FILL;
             }
+
             func_800788CC(NA_SE_EV_SAND_STORM - SFX_FLAG);
             break;
-        case 33:
+
+        case CS_MISC_SUNSSONG_START:
             gSaveContext.sunsSongState = SUNSSONG_START;
             break;
-        case 34:
+
+        case CS_MISC_FREEZE_TIME:
             if (IS_DAY) {
                 gSaveContext.dayTime -= gTimeSpeed;
             } else {
                 gSaveContext.dayTime -= gTimeSpeed * 2;
             }
             break;
```

Comment:
Outstanding implementation right there

wanna add a note this isn't exactly the mirror of
https://github.com/zeldaret/oot/blob/aa48c66e10fc067e4d41c3a1d097bd83dea53acf/src/code/z_kankyo.c#L927-L931
? lel